test/e2e/load_generator_test.go

```diff
@@ -25,30 +25,41 @@ import (
 	"github.com/knative/test-infra/shared/loadgenerator"
 )
 
-func loadTest(t *testing.T, factors []float64) {
+func loadTest(t *testing.T, factors bool, profiler bool) {
 	opts := loadgenerator.GeneratorOptions{
 		URL:            "http://www.google.com",
 		Duration:       10 * time.Second,
 		BaseQPS:        10,
 		NumThreads:     1,
 		NumConnections: 1,
 		RequestTimeout: 10 * time.Second,
-		LoadFactors:    factors,
+	}
+
+	if factors {
+		opts.LoadFactors = []float64{1, 2, 4}
+	}
+
+	if profiler {
+		opts.FileNamePrefix = t.Name()
 	}
 
 	res, err := opts.RunLoadTest(true)
 	if err != nil {
 		t.Fatalf("Error performing load test: %v", err)
 	}
 
-	if len(res.Result) != len(factors) {
-		t.Logf("got:%d, want: %d", len(res.Result), len(factors))
+	if factors && len(res.Result) != 3 {
+		t.Logf("got:%d, want: 3", len(res.Result))
```

Comment:
should this fail test? if not can you add a comment here explaining why this should not fail test?

Reply:
yup it should. Updated